# AirBnB NY Locations Data Case Study

In this final project, you task will be to take the data provided and find evidance to answer the following questions.

1. Which hosts are the busiest and why?
2. How many neighborhood groups are available and which shows up the most?
3. Are private rooms the most popular in manhattan?
4. Which hosts are the busiest and based on their reviews?
5. Which neighorhood group has the highest average price?
6. Which neighborhood group has the highest total price?
7. Which top 5 hosts have the highest total price?
8. Who currently has no (zero) availability with a review count of 100 or more?
9. What host has the highest total of prices and where are they located?
10. When did Danielle from Queens last receive a review?

You will be given **4 hours** to complete this assignment. 
**Be Advised** I will go dark for this intire assignment time period. That said, any questions that you would like to ask about the data, or the project **MUST** be asked before the time starts. Once the time has started, I can no longer give information.

This is to similate what you will face when you are out in the wild. 

Happy Coding!

In [2]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

In [3]:
air_bnb = pd.read_csv(r'C:\Users\kevin\OneDrive\Desktop\Coding Temple\Week 3 (Python)\AB_NYC_2019 (1).csv')
air_bnb.head()

id                                              name  host_id  \
0  2539                Clean & quiet apt home by the park     2787   
1  2595                             Skylit Midtown Castle     2845   
2  3647               THE VILLAGE OF HARLEM....NEW YORK !     4632   
3  3831                   Cozy Entire Floor of Brownstone     4869   
4  5022  Entire Apt: Spacious Studio/Loft by central park     7192   

     host_name neighbourhood_group neighbourhood  latitude  longitude  \
0         John            Brooklyn    Kensington  40.64749  -73.97237   
1     Jennifer           Manhattan       Midtown  40.75362  -73.98377   
2    Elisabeth           Manhattan        Harlem  40.80902  -73.94190   
3  LisaRoxanne            Brooklyn  Clinton Hill  40.68514  -73.95976   
4        Laura           Manhattan   East Harlem  40.79851  -73.94399   

         room_type  price  minimum_nights  number_of_reviews last_review  \
0     Private room    149               1                  9  2018-10-19   
1  Entire home/apt    225               1                 45  2019-05-21   
2     Private room    150               3                  0         NaN   
3  Entire home/apt     89               1                270  2019-07-05   
4  Entire home/apt     80              10                  9  2018-11-19   

   reviews_per_month  calculated_host_listings_count  availability_365  
0               0.21                               6               365  
1               0.38                               2               355  
2                NaN                               1               365  
3               4.64                               1               194  
4               0.10                               1                 0

In [4]:
# How many neighborhood groups are available and which shows up the most?
df = pd.read_csv(r'C:\Users\kevin\OneDrive\Desktop\Coding Temple\Week 3 (Python)\AB_NYC_2019 (1).csv')
unique_values =df['neighbourhood_group'].unique()
print("The unique values for neighborhood group are:")
print(unique_values)
num_unique_values = df['neighbourhood_group'].nunique()
print(f"The number of neighborhoods is {num_unique_values}")
value_counts = df['neighbourhood_group'].value_counts()
print("\nBelow is how many times each neighborhood is listed:")
print(value_counts)


The unique values for neighborhood group are:
['Brooklyn' 'Manhattan' 'Queens' 'Staten Island' 'Bronx']
The number of neighborhoods is 5

Below is how many times each neighborhood is listed:
Manhattan        21661
Brooklyn         20104
Queens            5666
Bronx             1091
Staten Island      373
Name: neighbourhood_group, dtype: int64


In [5]:
# Are private rooms the most popular in manhattan?

value_counts = df['room_type'].value_counts()
print("\nBelow is how many times each room type is listed:")
print(value_counts)
print("\nIn Manhattan specifically, the room types are:")
manhattan_df = df[df['neighbourhood_group'] == 'Manhattan']
room_type_counts = manhattan_df['room_type'].value_counts()
print(room_type_counts)
print("\nThis shows that an Entire home/apt is the most popular.")


Below is how many times each room type is listed:
Entire home/apt    25409
Private room       22326
Shared room         1160
Name: room_type, dtype: int64

In Manhattan specifically, the room types are:
Entire home/apt    13199
Private room        7982
Shared room          480
Name: room_type, dtype: int64

This shows that an Entire home/apt is the most popular.


In [14]:
# Which hosts are the busiest and based on their reviews?
df = pd.read_csv(r'C:\Users\kevin\OneDrive\Desktop\Coding Temple\Week 3 (Python)\AB_NYC_2019 (1).csv')
host_review_sums = df.groupby('host_name')['number_of_reviews'].sum()
host_with_most_reviews = host_review_sums.idxmax()
most_reviews = host_review_sums.max()
print(f"the host with the most reviews, and therefore the most popular is {host_with_most_reviews} with {most_reviews} reviews.")

host_review_sums = df.groupby('host_name')['number_of_reviews'].sum()
busiest_hosts = host_review_sums.nlargest(5)
print("The 5 busiest hosts are:")
for host, reviews in busiest_hosts.iteritems():
    print(f"{host} with {reviews} reviews.")
print("\n")
df["reviews_per_month"] = pd.to_datetime(df["reviews_per_month"])

reviews_per_month = df.groupby(["host_name", pd.Grouper(key="reviews_per_month", freq="M")])["number_of_reviews"].mean()

busiest_hosts = reviews_per_month.groupby("host_name").mean().nlargest(5)

print("The hosts with the highest mean number of reviews per month are:")
for host, mean_reviews_per_month in busiest_hosts.iteritems():
    print(f"{host} with {mean_reviews_per_month:.2f} reviews per month.")


the host with the most reviews, and therefore the most popular is Michael with 11081 reviews.
The 5 busiest hosts are:
Michael with 11081 reviews.
David with 8103 reviews.
John with 7223 reviews.
Jason with 6522 reviews.
Alex with 6204 reviews.


The hosts with the highest mean number of reviews per month are:
Dona with 602.50 reviews per month.
Asa with 488.00 reviews per month.
Dennis & Naoko with 441.00 reviews per month.
Miss Dy with 434.00 reviews per month.
Shunichi with 430.00 reviews per month.


C:\Users\kevin\AppData\Local\Temp\ipykernel_7432\2329955453.py:11: FutureWarning: iteritems is deprecated and will be removed in a future version. Use .items instead.
  for host, reviews in busiest_hosts.iteritems():
C:\Users\kevin\AppData\Local\Temp\ipykernel_7432\2329955453.py:21: FutureWarning: iteritems is deprecated and will be removed in a future version. Use .items instead.
  for host, mean_reviews_per_month in busiest_hosts.iteritems():


In [ ]:
#Which neighorhood group has the highest average price?
grouped_mean_prices = df.groupby('neighbourhood_group')['price'].mean()
highest_avg_price_group = grouped_mean_prices.idxmax()
highest_avg_price = grouped_mean_prices.max()
print(f"{highest_avg_price_group} is the neighborhood with the highest average price, the average price being ${highest_avg_price}.")

Manhattan is the neighborhood with the highest average price, the average price being $196.8758136743456.


In [34]:
# Which neighbor hood group has the highest total price?
grouped_total_prices = df.groupby('neighbourhood_group')['price'].sum()
highest_total_price_group = grouped_total_prices.idxmax()
highest_total_price = grouped_total_prices.max()
print(f"The neighborhood with the higest total price is {highest_total_price_group} with a total price of ${highest_total_price}.")

The neighborhood with the higest total price is Manhattan with a total price of $4264527.


In [43]:
#Which top 5 hosts have the highest total price?
import pandas as pd
df = pd.read_csv(r'C:\Users\kevin\OneDrive\Desktop\Coding Temple\Week 3 (Python)\AB_NYC_2019 (1).csv')
grouped_total_prices = df.groupby('host_name')['price'].sum()
top_5_hosts = grouped_total_prices.nlargest(5)
print("The top 5 hosts with the highest prices are:")
display(top_5_hosts)

The top 5 hosts with the highest prices are:


host_name
Sonder (NYC)    82795
Blueground      70331
Michael         66895
David           65844
Alex            52563
Name: price, dtype: int64

In [4]:
# Who currently has no (zero) availability with a review count of 100 or more?
import pandas as pd
df = pd.read_csv(r'C:\Users\kevin\OneDrive\Desktop\Coding Temple\Week 3 (Python)\AB_NYC_2019 (1).csv')

filtered_df = df[(df['availability_365'] == 0) &(df['number_of_reviews'] >= 100)]

print("Hosts without any availability who also have a review count of 100 or more are:")

filtered_host_names = filtered_df['host_name'].to_list()
print(filtered_host_names)

Hosts without any availability who also have a review count of 100 or more are:
['MaryEllen', 'Christiana', 'Sol', 'Coral', 'Doug', 'Ori', 'Lissette', 'Liz And Melissa', 'Ivy', 'Jsun', 'Wanda', 'Ben', 'S', 'Adrienne', 'Lydia', 'Karin', 'Elle', 'James', 'Jon', 'Liz', 'Jeanine', 'Lorena', 'Ron', 'Dragan', 'Misty', 'Brian', 'Natalie', 'AJ And Freddy', 'Neil & Katie', 'Emily', 'Evelyn', 'Alvaro', 'Bernard', 'Sarah', 'Karen', 'Summer', 'William', 'Andy & Friends', 'Karece', 'Andy & Friends', 'Ehren', 'Nicole', 'Terri', 'Ravanna', 'Molly', 'Lane', 'Angelo', 'DeLex', 'Michelle', 'Katarina', 'Véronique', 'Andreas', 'Caroline', 'Michael', 'Aurea', 'Kent', 'Brendan', 'Jillian', 'Deanna', 'Jake', 'Emily And Joel', 'Brian', 'George & Diana', 'Veronica', 'Masha', 'William', 'Danielle', 'Jeremy', 'Kyle', 'Stacey', 'Sasha', 'Nick', 'Carlina', 'Taylor & Tee', 'Devin', 'Ame', 'Richard', 'Micah', 'Elliott', '正川', 'Chao', 'Pj', 'Lou', 'Ingrid', 'Emily And Joel', 'Graham', 'Gurpreet  Singh', 'Gurpreet  Si

In [8]:
# What host has the highest total of prices and where are they located?
import pandas as pd
df = pd.read_csv(r'C:\Users\kevin\OneDrive\Desktop\Coding Temple\Week 3 (Python)\AB_NYC_2019 (1).csv')

grouped_total_prices = df.groupby('host_id')['price'].sum()
host_highest_price = grouped_total_prices.idxmax()
host_info = df[df['host_id'] == host_highest_price]
host_name = host_info['host_name'].iloc[0]
highest_total_price = grouped_total_prices[host_highest_price]
locations = host_info['neighbourhood'].unique()

print(f"The host with the highest total price is '{host_name}' (host ID: {host_highest_price}), with a total price of ${highest_total_price}.")
print(f"They are located in {', '.join(locations)}.")

The host with the highest total price is 'Sonder (NYC)' (host ID: 219517861), with a total price of $82795.
They are located in Financial District, Murray Hill, Hell's Kitchen, Midtown, Theater District, Chelsea, Upper East Side.


In [7]:
# When did Danielle from Queens last receive a review?
import pandas as pd

df = pd.read_csv(r'C:\Users\kevin\OneDrive\Desktop\Coding Temple\Week 3 (Python)\AB_NYC_2019 (1).csv')

df['last_review'] = pd.to_datetime(df['last_review'], errors='coerce')

danielle_review = df[(df['host_name'] == 'Danielle') & (df['neighbourhood_group'] == 'Queens')]

last_review_date = danielle_review['last_review'].max()

print(f"The last time Danielle from Queens received a review was on {last_review_date}.")

The last time Danielle from Queens received a review was on 2019-07-08 00:00:00.


## Further Questions

1. Which host has the most listings?

In [8]:
import pandas as pd

df = pd.read_csv(r'C:\Users\kevin\OneDrive\Desktop\Coding Temple\Week 3 (Python)\AB_NYC_2019 (1).csv')

listing_counts = df.groupby('host_id').size()
host_most_listings = listing_counts.idxmax()
host_info = df[df['host_id'] == host_most_listings].iloc[0]
host_name = host_info['host_name']
num_listings = listing_counts[host_most_listings]
print (f"{host_name}, id: {host_most_listings} has the most listings with {num_listings} listings.")

Sonder (NYC), id: 219517861 has the most listings with 327 listings.


2. How many listings have completely open availability?

In [9]:
import pandas as pd
df = pd.read_csv(r'C:\Users\kevin\OneDrive\Desktop\Coding Temple\Week 3 (Python)\AB_NYC_2019 (1).csv')
completely_open = df[df['availability_365'] == 365]
num_completely_open = len(completely_open)
print(f"There are {num_completely_open} listings with completely open availability.")

There are 1295 listings with completely open availability.


3. What room_types have the highest review numbers?

In [13]:
import pandas as pd
df = pd.read_csv(r'C:\Users\kevin\OneDrive\Desktop\Coding Temple\Week 3 (Python)\AB_NYC_2019 (1).csv')
grouped_reviews = df.groupby('room_type')['number_of_reviews'].sum()
highest_room_type_review = grouped_reviews[grouped_reviews == grouped_reviews.max()]
print(f"The room type with the highest reviews is: {highest_room_type_review}")

The room type with the highest reviews is: room_type
Entire home/apt    580403
Name: number_of_reviews, dtype: int64


# Final Conclusion

In this cell, write your final conclusion for each of the questions asked.

Also, if you uncovered some more details that were not asked above, please discribe them here.

-- Add your conclusion --

1. Which hosts are the busiest and why?

I picked these hosts because they are the top 5 with the most reviews per month:

Dona with 602.50 reviews per month.
Asa with 488.00 reviews per month.
Dennis & Naoko with 441.00 reviews per month.
Miss Dy with 434.00 reviews per month.
Shunichi with 430.00 reviews per month.

I picked the busiest hosts based on which 5 hosts had the top reviews.

Michael had 11081 reviews.
David had 8103 reviews.
John had 7223 reviews.
Jason had 6522 reviews.
Alex had 6204 reviews.

2. How many neighborhood groups are available and which shows up the most?

There are 5 neighborhood groups availalble with Manhattan showing up the most.

3. Are private rooms the most popular in manhattan?

The most popular room type in Manhattan is an Entire home/apt with a count of 13199.

4. Which hosts are the busiest and based on their reviews?

I picked the busiest hosts based on which 5 hosts had the top reviews.

Michael had 11081 reviews.
David had 8103 reviews.
John had 7223 reviews.
Jason had 6522 reviews.
Alex had 6204 reviews.

5. Which neighorhood group has the highest average price?

Manhattan has the highest average price, the average price being $197.00


6. Which neighborhood group has the highest total price?

Manhattan has the highest total price which is $4,264,527.

7. Which top 5 hosts have the highest total price?

The top 5 hosts with the highest total prices are below along with their total prices.

Sonder (NYC)    $82,795
Blueground      $70,331
Michael         $66,895
David           $65,844
Alex            $52,563

8. Who currently has no (zero) availability with a review count of 100 or more?

There are a lot of hosts wtih a review count of 100 or more who don't have any availabity. They are in the below list:

['MaryEllen', 'Christiana', 'Sol', 'Coral', 'Doug', 'Ori', 'Lissette', 'Liz And Melissa', 'Ivy', 'Jsun', 'Wanda', 'Ben', 'S', 'Adrienne', 'Lydia', 'Karin', 'Elle', 'James', 'Jon', 'Liz', 'Jeanine', 'Lorena', 'Ron', 'Dragan', 'Misty', 'Brian', 'Natalie', 'AJ And Freddy', 'Neil & Katie', 'Emily', 'Evelyn', 'Alvaro', 'Bernard', 'Sarah', 'Karen', 'Summer', 'William', 'Andy & Friends', 'Karece', 'Andy & Friends', 'Ehren', 'Nicole', 'Terri', 'Ravanna', 'Molly', 'Lane', 'Angelo', 'DeLex', 'Michelle', 'Katarina', 'Véronique', 'Andreas', 'Caroline', 'Michael', 'Aurea', 'Kent', 'Brendan', 'Jillian', 'Deanna', 'Jake', 'Emily And Joel', 'Brian', 'George & Diana', 'Veronica', 'Masha', 'William', 'Danielle', 'Jeremy', 'Kyle', 'Stacey', 'Sasha', 'Nick', 'Carlina', 'Taylor & Tee', 'Devin', 'Ame', 'Richard', 'Micah', 'Elliott', '正川', 'Chao', 'Pj', 'Lou', 'Ingrid', 'Emily And Joel', 'Graham', 'Gurpreet  Singh', 'Gurpreet  Singh', 'Jimmy', 'Catrina', 'Long', 'Deborah', 'Hayes', 'Hayes', 'Evan', 'Sofia', 'Sofia', 'Antonia', 'Margarita', 'Abraham', 'Alex', 'Qiyao', 'Cedrick', 'Greg', 'Chelsea', 'Kc', 'Edward', 'Lasata', 'Krysta', 'Maeve', 'Erin', 'Trevor', 'Giovanni', 'Anthony', 'Chadanut', 'Dante', 'Braydon', 'Gillian', 'Fred', 'Maba', 'Steven', 'Marisha', 'Alex', 'Felipe', 'Airamis', 'Anna', 'Michael', 'Michael', 'Michael', 'Raymond', 'Michael', 'Gabriela', 'Gabriela', 'Kaka', 'Benjamin', 'Jo & Savannah', 'Jo & Savannah', 'J', 'Alexandra', 'Mohamed', 'Mohamed', 'Deloris', 'Deloris', 'Megan E', 'Brian', 'Gurpreet  Singh', 'Gurpreet  Singh', 'Yuval', 'Marcelo', 'Jie', 'Brendan', 'Sam', 'Andrew', 'Ash', 'Jenifer', 'Louise', nan, 'Kathleen', 'Janet', 'Albert', 'Stephany', 'Mariluz']

9. What host has the highest total of prices and where are they located?

The host with the highest total price is 'Sonder (NYC), with a total price of $82795.
They are located in the Financial District, Murray Hill, Hell's Kitchen, Midtown, the Theater District, Chelsea, and the Upper East Side.


10. When did Danielle from Queens last receive a review?

The last time Danielle from Queens received a review was on 2019-07-08.